### PR TITLE
BAVL-71 adding the necessary client for making calls to the activities and appointments API.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
This change will allow our service to make calls to the external activities and appointments API for dealing with video link appointments between the two services.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/wiremock/ActivitiesAppointmentsApiMockServer.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  fun stubGetAppointment(appointmentId: Long) {
+    stubFor(
+      WireMock.get("/appointments/$appointmentId")
+        .willReturn(
+          WireMock.aResponse()
+            .withHeader("Content-Type", "application/json")
+            .withBody(
+              mapper.writeValueAsString(
+                Appointment(
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/wiremock/ActivitiesAppointmentsApiMockServer.kt:24]: We can pull this out into a factory class as and when necessary.

File: build.gradle.kts [hunk 2]
```diff
@@ -72,15 +72,24 @@ val configValues = mapOf(
 val buildDirectory: Directory = layout.buildDirectory.get()
 
 tasks.register("buildLocationsInsidePrisonApiModel", GenerateTask::class) {
-  generatorName.set("kotlin-spring")
+  generatorName.set("kotlin")
```
Comment (matt-crowson-moj) [inline build.gradle.kts:75]: Changed as kotlin-spring was generating OffsetDateTime fields when what we actually want is LocalDateTime fields.

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -38,6 +38,12 @@ spring:
           jwk-set-uri: ${api.base.url.hmpps-auth}/.well-known/jwks.json
       client:
         registration:
+          activities-appointments:
+            provider: hmpps-auth
+            client-id: ${system.client.id}
+            client-secret: ${system.client.secret}
+            authorization-grant-type: client_credentials
+            scope: read
```
Comment (timharrison-moj) [inline src/main/resources/application.yml:46]: Probably will need to be read/write here - but fine for now.

**Review by timharrison-moj — APPROVED** [review]
LGTM.

File: src/main/resources/application.yml (hunk 3 above)
Comment (matt-crowson-moj) [reply src/main/resources/application.yml:46]: Good spot.

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**